cmip6_downscaling/methods/bcsd.py

```diff
@@ -168,25 +177,37 @@ def get_spatial_anomalies(
     seasonal_cycle_spatial_anomalies : xr.Dataset
         Spatial anomaly for each month (i.e. of shape (nlat, nlon, 12))
     """
-    # Regrid coarse observation dataset to the spatial scale of the raw obs
-    coarse_obs_interpolated, _ = regrid_dataset(
-        ds=coarse_obs,
-        ds_path=None,
-        target_grid_ds=obs_ds.isel(time=0),
-        variable=variable,
+    print(kwargs)
+
+    interpolated_obs_path = (
+        config.get("storage.intermediate.uri")
+        + '/'
+        + make_interpolated_obs_path(
+            gcm_grid_spec=gcm_grid_spec,
+            chunking_approach=chunking_approach,
+            obs_identifier=obs_identifier,
+        )
+    )
+    coarse_obs_interpolated_rechunked_path = rechunk_zarr_array_with_caching(
+        interpolated_obs_path, chunking_approach='full_time', max_mem='2GB'
+    )
+
+    obs_path = (
+        config.get("storage.intermediate.uri")
+        + '/'
+        + make_return_obs_path(obs_identifier=obs_identifier)
     )
-    obs_rechunked, _ = rechunk_zarr_array(
-        obs_ds,
-        None,
-        chunk_dims=('time',),
-        variable=variable,
-        max_mem='1GB',
+    obs_rechunked_path = rechunk_zarr_array_with_caching(
+        obs_path, chunking_approach='full_time', max_mem='2GB'
     )
-    # calculate the difference between the actual obs (with finer spatial heterogeneity)
-    # and the interpolated coarse obs this will be saved and added to the
-    # spatially-interpolated coarse predictions to add the spatial heterogeneity back in.
+    coarse_obs_interpolated_rechunked_ds = xr.open_zarr(coarse_obs_interpolated_rechunked_path)
+    obs_rechunked_ds = xr.open_zarr(obs_rechunked_path)
+
+    # # calculate the difference between the actual obs (with finer spatial heterogeneity)
+    # # and the interpolated coarse obs this will be saved and added to the
+    # # spatially-interpolated coarse predictions to add the spatial heterogeneity back in.
```

Comment:
```suggestion
    # calculate the difference between the actual obs (with finer spatial heterogeneity)
    # and the interpolated coarse obs this will be saved and added to the
    # spatially-interpolated coarse predictions to add the spatial heterogeneity back in.
```